test/common/test_executor.cpp

```diff
@@ -0,0 +1,250 @@
+
+  EXPECT_TRUE((is_instance_of_any_v<DerivedExecutor, inline_executor, omp_executor>));
+  EXPECT_FALSE((is_instance_of_any_v<DerivedExecutor, sse_executor, omp_executor>));
+}
+
+TEST(ExecutorRuntimeChecks, ExecutorEnviornmentVariableCheck)
+{
+  auto env_check = [&](auto exec, const char* env_name) {
+    setenv(env_name, "OFF", true);
+    EXPECT_FALSE(executor_runtime_checks::check(exec));
+    setenv(env_name, "ON", true);
+    EXPECT_TRUE(executor_runtime_checks::check(exec));
+  };
+
+  env_check(default_sse_executor{}, "PCL_ENABLE_SSE_EXEC");
+  env_check(default_omp_executor{}, "PCL_ENABLE_OMP_EXEC");
+}
+
+class MatrixMultiplication {
+protected:
+  template <
+      typename ExecutorT,
+      typename executor::InstanceOfAny<ExecutorT, inline_executor, sse_executor> = 0>
+  void
+  mmul(const ExecutorT ex,
+       const Eigen::MatrixXd& a,
+       const Eigen::MatrixXd& b,
+       Eigen::MatrixXd& c)
+  {
+    auto mul = [&]() { c = a * b; };
+    ex.execute(mul);
+  }
+
+  template <typename ExecutorT,
+            typename executor::InstanceOf<ExecutorT, omp_executor> = 0>
+  void
+  mmul(const ExecutorT ex,
+       const Eigen::MatrixXd& a,
+       const Eigen::MatrixXd& b,
+       Eigen::MatrixXd& c)
+  {
+    auto mul = [&](typename decltype(ex)::index_type index) {
+#pragma omp for schedule(static)
+      for (int i = 0; i < a.rows(); i = i + 1) {
+        for (int j = 0; j < b.cols(); j = j + 1) {
+          c(i, j) = 0.0;
+          for (int k = 0; k < a.cols(); k = k + 1) {
+            c(i, j) += a(i, k) * b(k, j);
+          }
+        }
+      }
+    };
+    ex.bulk_execute(mul, a.rows());
+  }
+
+  void
+  mmul(const Eigen::MatrixXd& a,
+       const Eigen::MatrixXd& b,
+       Eigen::MatrixXd& c,
+       bool custom_priority)
+  {
+    auto mul = [&](auto& exec) { mmul(exec, a, b, c); };
+
+    auto supported_executors =
+        std::conditional_t<is_executor_available_v<omp_executor>,
+                           std::tuple<default_omp_executor, default_inline_executor>,
+                           std::tuple<default_inline_executor>>();
+
+    if (custom_priority)
+      enable_exec_with_priority(mul, supported_executors);
+    else
+      enable_exec_on_desc_priority(mul, supported_executors);
+  }
+};
+
+template <typename Executor>
+class ExecutorMatrixMultiplication : public ::testing::Test,
+                                     public MatrixMultiplication {};
+
+TYPED_TEST_SUITE(ExecutorMatrixMultiplication, ExecutorTypes);
+
+TYPED_TEST(ExecutorMatrixMultiplication, executors)
+{
+  TypeParam exec;
+
+  double dataA[9] = {1, 2, 3, 4, 5, 6, 7, 8, 9}, dataB[9] = {1, 2, 3, 4, 5, 6, 7, 8, 9},
+         dataC[9] = {0};
+  Eigen::MatrixXd a = Eigen::Map<Eigen::Matrix<double, 3, 3, Eigen::RowMajor>>(dataA);
+  Eigen::MatrixXd b = Eigen::Map<Eigen::Matrix<double, 3, 3, Eigen::RowMajor>>(dataB);
+  Eigen::MatrixXd c = Eigen::Map<Eigen::Matrix<double, 3, 3, Eigen::RowMajor>>(dataC);
```

Comment:
The size of a, b and c are known at compile time.

Reply:
I am not quite sure what to do with this?

Reply:
Have you tried just doing this.
```suggestion
const Eigen::Map<const Eigen::Matrix<double, 3, 3, Eigen::RowMajor>> a(dataA);
const Eigen::Map<const Eigen::Matrix<double, 3, 3, Eigen::RowMajor>> b(dataB);
Eigen::Map<Eigen::Matrix<double, 3, 3, Eigen::RowMajor>> c(dataC);
```